src/lsp/cobol_lsp/lsp_completion.ml

```diff
@@ -13,91 +13,148 @@
 
 open EzCompat
 
-open Cobol_common                                                  (* Visitor *)
+open Cobol_common                                                  (* Visitor, NEL *)
 open Cobol_common.Srcloc.INFIX
 
-open Lsp_completion_keywords
 open Lsp.Types
 
-let name_proposals ast ~filename pos =
-  let visitor = object
-    inherit [StringSet.t] Cobol_ptree.Visitor.folder
-
-    method! fold_compilation_unit' cu =
-      if Lsp_position.is_in_srcloc ~filename pos ~@cu
-      then Visitor.do_children
-      else Visitor.skip_children
-
-    method! fold_name name acc =
-      Visitor.skip_children @@ StringSet.add name acc
-
-    end
+module Menhir = Cobol_parser.Grammar_interpr
+module Expect = Cobol_parser.Expect
+
+let qualname_proposals ~filename pos group =
```

Comment:
I think this should be `qualnames_proposal` (it is a proposal of multiple qualnames, if I understand it right)